.github/workflows/sonar.yml

```diff
@@ -44,14 +44,14 @@ jobs:
       - name: Download artifact from CI workflow
         uses: actions/download-artifact@v4
         with:
-          name: code-coverage-report
+          name: code-coverage-report-${{ matrix.python-version }}
```

Comment:
We are not running this one in a matrix no? We can, instead, use `pattern` to download all the matching reports into a directory, and then let sonarcloud pick up the reports

Reply:
The step uploading the artifacts is inside the matrix. Not sure if the Sonar analysis would be different between python versions.